src/driver/express/ExpressDriver.ts

```diff
@@ -222,25 +223,26 @@ export class ExpressDriver extends BaseDriver implements Driver {
     handleSuccess(result: any, action: ActionMetadata, options: Action): void {
 
         // check if we need to transform result and do it
-        if (this.useClassTransformer && result && result instanceof Object) {
+        if (result && result instanceof Object && this.useClassTransformer) {
             const options = action.responseClassTransformOptions || this.classToPlainTransformOptions;
             result = classToPlain(result, options);
         }
 
         // set http status code
-        if (action.undefinedResultCode && result === undefined) {
-            if (action.undefinedResultCode instanceof Function)
-                throw new (action.undefinedResultCode as any)(options);
-
-            options.response.status(action.undefinedResultCode);
```

Comment:
In this place I only left checking `undefinedResultCode`. However now it's not about `set http status code` so I will move it to the handler on the bottom.